azurerm/internal/services/containers/kubernetes_cluster_maintenance_configuration_resource.go

```diff
@@ -0,0 +1,277 @@
+package containers
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-03-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/parse"
+	containerValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceKubernetesClusterMaintenanceConfiguration() *pluginsdk.Resource {
```

Comment:
Is it possible to provision more than one of these against a cluster, if so can we add a test for this? Otherwise this resource should be embedded within the existing `azurerm_kubernetes_cluster` resource?

Reply:
@tombuildsstuff thanks for your suggestions. multiple maintenance configuration can be applied to one kubernetes cluster. I have added a test for it.

Reply:
We should chat with the AKS Team about this too, this feels like this design will change to match other resources where a single schedule is available

Reply:
@njuCZ @ms-henglu did you reach out to the AKS team to confirm their plans here?

Reply:
@tombuildsstuff , Yes, service team told me that this is by design.

Reply:
@ms-henglu by design as in they intend to support a single configuration, or that multiple configurations will/are supported?

Reply:
@tombuildsstuff , It should support multiple in the future, but currently only default works. 
refs : https://docs.microsoft.com/en-us/azure/aks/planned-maintenance#allow-maintenance-on-every-monday-at-100am-to-200am